executor/partition_table_test.go

```diff
@@ -305,6 +305,82 @@ func (s *partitionTableSuite) TestGlobalStatsAndSQLBinding(c *C) {
 	tk.MustIndexLookup("select * from tlist where a<1")
 }
 
+func (s *partitionTableSuite) TestDirectReadingwithJoin(c *C) {
```

Comment:
And what particular type of Join this test want to cover, HashJoin, IndexJoin, or MergeJoin?

Reply:
This test intends to test IndexJoin with different direct reading functors as its inner side.

I just found that IndexJoin might not have table reader functor as its inner side. How about we only test IndexLookup and IndexReader and delete TableReader?

Reply:
Done in the updated commit.